deepchem/models/torch_models/gcn.py

```diff
@@ -0,0 +1,336 @@
+"""
+DGL-based GCN for graph property prediction.
+"""
+import torch.nn as nn
+import torch.nn.functional as F
+
+from deepchem.models.losses import L2Loss, SparseSoftmaxCrossEntropy
+from deepchem.models.torch_models.torch_model import TorchModel
+
+class GCN(nn.Module):
+    """Model for Graph Property Prediction Based on Graph Convolution Networks (GCN).
+
+    This model proceeds as follows:
+
+    * Update node representations in graphs with a variant of GCN
+    * For each graph, compute its representation by 1) a weighted sum of the node
+      representations in the graph, where the weights are computed by applying a
+      gating function to the node representations 2) a max pooling of the node
+      representations 3) concatenating the output of 1) and 2)
+    * Perform the final prediction using an MLP
+
+    Examples
+    --------
+
+    >>> import deepchem as dc
+    >>> import pymatgen as mg
+    >>> from deepchem.models import GCN
+    >>> lattice = mg.Lattice.cubic(4.2)
+    >>> structure = mg.Structure(lattice, ["Cs", "Cl"], [[0, 0, 0], [0.5, 0.5, 0.5]])
+    >>> featurizer = dc.feat.CGCNNFeaturizer()
```

Comment:
It might be useful to have both featurizations as examples. I think DeepChem is starting to pick up more users from the materials science communities so both examples would be great :)

Reply:
I agreed that it is better to support both crystal graph (CGCNNFeaturizer) and molecule graph (MolGraphConvFeaturizer) ideally.

But, as I mentioned in https://github.com/deepchem/deepchem/pull/2249#discussion_r514288251, we should take care of some points for supporting them.

Reply:
I've replaced `CGCNNFeaturizer` with `MolGraphConvFeaturizer` in the examples.